src/scenes/CollectionGalleryPage/CollectionGalleryPage.tsx

```diff
@@ -32,6 +34,29 @@ function CollectionGalleryPage({ collectionId, username, queryRef }: CollectionG
     });
   }, [username, collectionId, track]);
 
+  const ePress = useKeyDown('e');
+  const escapePress = useKeyDown('Escape');
+  const backspacePress = useKeyDown('Backspace');
+
+  const { push } = useRouter();
+
+  const navigateToEdit = useCallback(() => {
+    void push(`/edit?collectionId=${collectionId}`);
```

Comment:
let's only navigate to the collection edit page for the given ID if the collection belongs to the user. when i'm on a collection page that i _don't_ own, it should not append the URL query param. 

current behavior:
- go to collection that i don't own: `/mikey/22PDnzYR9Xgj2lRRHrD3j7dNpVz`
- click "e"
- get navigated to `/edit?collectionId=22PDnzYR9Xgj2lRRHrD3j7dNpVz`, even though it's not my collection

desired behavior: clicking "e" should just navigate to my own edit page (i.e. /edit)